Movieee/Movieee/UIs/Home Screen/HomeScreenViewController.swift

```diff
@@ -0,0 +1,130 @@
+    
+    @IBAction func searchButtonPressed(_ sender: UIButton) {
+        let searchStoryBoard: UIStoryboard = UIStoryboard(name: "SearchScreen", bundle: nil)
+        guard let searchVC = searchStoryBoard.instantiateViewController(withIdentifier: "SearchScreenViewController") as? SearchScreenViewController else {
+            return
+        }
+        navigationController?.pushViewController(searchVC, animated: true)
+    }
+    
+    enum CollectionSection: Int {
+        case popular = 0, nowPlaying, upcoming, toprated
+        var valueConvert : String {
+            switch self {
+            case .popular : return ConstraintsHomeScreen.KindOfCategoryURL.popular.rawValue
+            case .nowPlaying : return ConstraintsHomeScreen.KindOfCategoryURL.nowPlaying.rawValue
+            case .upcoming : return ConstraintsHomeScreen.KindOfCategoryURL.upComing.rawValue
+            case .toprated: return ConstraintsHomeScreen.KindOfCategoryURL.topRated.rawValue
+            }
+        }
+    }
+    
+    private var data = [CollectionSection: [CategoryResults]]()
+    private let dispatchGroup = DispatchGroup()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        homeScreenTableView.delegate = self
+        homeScreenTableView.dataSource = self
+        
+        for index in 0..<ConstraintsHomeScreen.KindOfCategoryURL.allCases.count {
+            configHomeScreen(index: index,
+                             category: ConstraintsHomeScreen.KindOfCategoryURL.allCases[index].rawValue)
+        }
+        
+        dispatchGroup.notify(queue: .main) {
+            self.homeScreenTableView.reloadData()
+        }
+    }
+    
+    private func configHomeScreen(index: Int, category: String) {
+        dispatchGroup.enter()
+        APIMovie.apiMovie.getMovieFromCategory(from: category) { [unowned self] result in
+            DispatchQueue.main.async {
+                data[CollectionSection(rawValue: index) ?? CollectionSection.nowPlaying] = result.results
+                dispatchGroup.leave()
+            }
+        }
+    }
+}
+
+extension HomeScreenViewController: UITableViewDelegate,
+                                    UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return data.count
+    }
+    
+    func tableView(_ tableView: UITableView,
+                   numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView,
+                   cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell =
+                tableView.dequeueReusableCell(withIdentifier: ConstraintsHomeScreen.idHomeTableView,
+                                              for: indexPath) as? HomeScreenMovieListTableViewCell
+        else { return UITableViewCell() }
+        cell.configCollectionCell( list: data[CollectionSection(rawValue: indexPath.section)
```

Comment:
chỗ này dùng guard let cho rõ ràng đi em. Gộp vào với guard let ở trên
```suggestion
        cell.configCollectionCell(list: data[CollectionSection(rawValue: indexPath.section)
```